### ISSUE TITLE
Problem: `mysql.host` is ignored

### ISSUE THREAD

**Issue by Sergio-IME** [ansible-semaphore/semaphore#2466]
### Issue

When provided with a complete `config.json` without any additional ENV vars, the `mysql.host` value is ignored and the default value `0.0.0.0` for `SEMAPHORE_DB_HOST` is used and thus the db connection fails.

When `SEMAPHORE_DB_HOST` is set to the correct host, it works, no matter the value of `mysql.host`.

### Impact

Configuration, Database

### Installation method

Docker

### Database

MySQL

### Semaphore Version

2.10.22+ (not tested before)

### Logs & errors

```console
semaphore-1  | Attempting to connect to database semaphore on 0.0.0.0:3306 with user semaphore ...
semaphore-1  | .............................
semaphore-1 exited with code 1
```

### Configuration

`compose.yml`
```yaml
services:
  semaphore:
    image: semaphoreui/semaphore:latest
    volumes:
      - ./config.json:/etc/semaphore/config.json
    #environment:
    # does not work without
    #  SEMAPHORE_DB_HOST: mysql

  mysql:
    image: mysql:8.0-debian
    volumes:
      - ./semaphore-mysql:/var/lib/mysql
    environment:
      MYSQL_RANDOM_ROOT_PASSWORD: 'yes'
      MYSQL_DATABASE: semaphore
      MYSQL_USER: semaphore
      MYSQL_PASSWORD: semaphore
```

`config.json`
```json
{
 	"mysql": {
 		"host": "mysql",
 		"user": "semaphore",
 		"pass": "semaphore",
 		"name": "semaphore",
 		"options": null
 	},
 	"dialect": "mysql",
 	"port": "",
 	"interface": "",
 	"tmp_path": "/tmp/semaphore",
 	"ssh_config_path": "",
 	"git_client": "",
 	"web_host": "https://semaphore.local",
 	"cookie_hash": "xxxx",
 	"cookie_encryption": "xxxx",
 	"access_key_encryption": "xxxx",
 	"email_alert": false,
 	"email_sender": "",
 	"email_host": "",
 	"email_port": "",
 	"email_username": "",
 	"email_password": "",
 	"email_secure": false,
 	"ldap_enable": false,
 	"ldap_binddn": "",
 	"ldap_bindpassword": "",
 	"ldap_server": "",
 	"ldap_searchdn": "",
 	"ldap_searchfilter": "",
 	"ldap_mappings": {
 		"dn": "",
 		"mail": "",
 		"uid": "",
 		"cn": ""
 	},
 	"ldap_needtls": false,
 	"telegram_alert": false,
 	"telegram_chat": "",
 	"telegram_token": "",
 	"slack_alert": false,
 	"slack_url": "",
 	"rocketchat_alert": false,
 	"rocketchat_url": "",
 	"microsoft_teams_alert": false,
 	"microsoft_teams_url": "https://teams/webhook",
 	"oidc_providers": null,
 	"max_task_duration_sec": 0,
 	"max_tasks_per_template": 0,
 	"max_parallel_tasks": 0,
 	"runner_registration_token": "",
 	"password_login_disable": false,
 	"non_admin_can_create_project": false,
 	"use_remote_runner": false,
 	"runner": {
 		"api_url": "",
 		"registration_token": "",
 		"config_file": "",
 		"one_off": false,
 		"webhook": "",
 		"max_parallel_tasks": 0
 	},
 	"global_integration_alias": "",
 	"apps": null
}
```

**Comment by fiftin** [comment]
Hi, fixed in v2.10.33. Please check.

**Comment by strenker** [comment]
Hi,

i just opened issue [#2502](https://github.com/semaphoreui/semaphore/issues/2502) which maybe could be related to this one.
